code/game/atoms.dm

```diff
@@ -1384,7 +1384,17 @@
 /atom/proc/log_message(message, message_type, color=null, log_globally=TRUE)
 	if(!log_globally)
 		return
-
+	if(CONFIG_GET(flag/sql_game_log) && CONFIG_GET(flag/sql_enabled))
+		//Build SQL Query to insert log into the DB, and catch errors
+		var/datum/db_query/query_sql_log_messages = SSdbcore.NewQuery({"
+			INSERT INTO [format_table_name("game_log")] (datetime, round_id, ckey, loc, type, message)
+			VALUES (:time, :round_id, :ckey, :loc, :type, :message)
+		"}, list("time" = SQLtime(), "round_id" = "[GLOB.round_id]", "ckey" = key_name(src), "loc" = loc_name(src), type = message_type, "message" = message))
+		if(!query_sql_log_messages.warn_execute())
+			qdel(query_sql_log_messages)
+			return
+		qdel(query_sql_log_messages)
+		return //Dont do file log when SQL log is enabled, for that is the big dumb.
```

Comment:
This shouldn't be an assumption you make. This should be a config option, if at all.

Reply:
Writing to both filesystem and SQL. is silly, and hurts performance in my testing. again from Disk IOPS

Reply:
Throwing away the log file is silly when you are only putting one in the database. I am one of several people who consume the text versions, and being unable to reference them without some (yet unproposed?) application is a fairly big problem. It's not silly.

Reply:
Ok I thought you were only logging game.txt lines, not all messages. With that covered I still am concerned about the dropping of file-based logs, and I would propose having a config instead of assuming this behaviour.

Reply:
Sure, I can add it as an option rather than just assuming to disable the file logs.
I have concerns that with both on, it would hurt performance. but i have not tested both running.